ansible/lifecycle_ec2.yml

```diff
@@ -4,32 +4,53 @@
     AWS_SECRET_ACCESS_KEY: "{{aws_secret_access_key}}"
     AWS_DEFAULT_REGION: "{{aws_region_final|d(aws_region)}}"
   block:
-    - when: ACTION == 'stop'
-      name: Stop instances by (guid,env_type) tags
+    - name: Get EC2 instances filtering by guid
+      ec2_instance_info:
+        filters:
+          "tag:guid": "{{ guid }}"
+      register: r_instances
+
+    - when: r_instances.instances | default([]) | length == 0
+      name: Fail if no instance is found
+      fail:
+        msg: "No instance found for guid={{ guid }}"
```

Comment:
Is it really a failure to have no instances? We might want to stop instances in a sandbox open environment and some sandboxes may have no instances?